qiskit/providers/ibmq/api_v2/ibmqclient.py

```diff
@@ -69,25 +69,27 @@ def _request_access_token(self):
 
         Returns:
             str: access token.
+
         Raises:
             AuthenticationLicenseError: if the user hasn't accepted the license agreement.
+            KeyError: if the response from the login attempt did not contain the expected JSON.
+
         """
         try:
-            self.client_auth.login(self.api_token)
+            response = self.client_auth.login(self.api_token)
+            return response['id']
         except RequestsApiError as ex:
             response = ex.original_exception.response
-            if response and response.status_code == 401:
+            if response is not None and response.status_code == 401:
                 try:
                     error_code = response.json()['error']['name']
                     if error_code == 'ACCEPT_LICENSE_REQUIRED':
                         message = response.json()['error']['message']
                         raise AuthenticationLicenseError(message)
-                except (ValueError, KeyError):
+                except KeyError as invalid_key:
                     # the response did not contain the expected json.
-                    pass
-
-        response = self.client_auth.login(self.api_token)
-        return response['id']
+                    raise KeyError('Invalid key access attempt in the JSON response '
```

Comment:
This is actually the exception that we should _not_ be raising - it only happens during the specific block that attempts to check for the license error.